Add ViewPanner to 2D editor

KoBeWi opened the pull request:
Summary of changes:
- Added ViewPanner to 2D editor
  - as a result, in default control scheme Ctrl+Wheel will scroll the view instead of zooming. But it duplicated the no-Ctrl wheel, so whatevs
- Added ViewPanner to TextureRegionEditor (I missed it in the last PR; some more editors could still be changed probably)

ViewPanner:
- added customizable panning shortcut, instead of hard-coded Space
- changed "disable RMB" to "enable RMB". RMB panning is now disabled by default as only 1 out of 8 editors uses it lol
- zoom callback got a new `p_alt` parameter, as some editors used "alternate zoom", so I simplified it
- simplified the callback helper method
- added "simple panning" support, i.e. spacebar panning without LMB
- added a `setup()` method that setups multiple common properties from editor
- I adjusted various editors due to these changes

Editor Settings:
- moved all panning-related settings to a dedicated "Panning" category
- `editors/2d/warped_mouse_panning` changed to `editors/panning/warped_mouse_panning`
  - now used by all GraphEdit-based editors
- `editors/2d/simple_panning` changed to `editors/panning/simple_panning`
  - now used by every editor with ViewPanner
- `editors/2d/pan_speed` changed to `interface/editors/2d_editor_pan_speed`
- `editors/2d/scroll_to_pan` changed to `interface/editors/2d_editor_panning_scheme`
- `canvas_item_editor/pan_view` shortcut is now supported by every editor with ViewPanner (maybe it could be moved?)

![image](https://user-images.githubusercontent.com/2223172/150358769-d68c0bed-f89c-474e-8bf9-7a151e00f2e3.png)

GraphEdit:
- restored mouse warping (forgot to add it in the first PR)
- added two unexposed methods: `get_panner()` and `set_warped_panning()`, used by editor

EDIT:
After most recent changes, it fixes #34278

KoBeWi committed:
Add ViewPanner to 2D editor

FireForger commented:
Have you changed the Sprite converter menu to use this yet? If not, you should definitely add that to your to-do list because it doesn't support zooming in yet.
![image](https://user-images.githubusercontent.com/67974470/150411959-adf36905-fe12-4b34-8e35-4662a0dbe96d.png)

KoBeWi commented:
Ok, I managed to fix the pan key issue. I also fixed a few 2D editor bugs related to panning.

- This thing (happens with simple panning):
![godot windows opt tools 64_aHQhGMJmZU](https://user-images.githubusercontent.com/2223172/150573823-901810a8-ef95-42ac-adaa-5e2c9c7b55f3.gif)
FIXED
- Selection box not disappearing when you hold Space (regression introduced back in #27770) FIXED
- #34278 FIXED
- Some stuff I probably forgot. I should've written it down .-.

Now every editor with ViewPanner can grab click focus to make shortcut working. I connected their `focus_exited` signal to the newly added `release_pan_key()` method. Unfortunately it doesn't work in the animation editor, but it wasn't available before, so whateves.

It would be nice if someone could test if panning works as expected (with different settings) in these editors:
- main 2D editor
- TextureRegion editor (e.g. in Sprite)
- Polygon2D UV editor
- TileSet tile editor
- AnimationTrack editor (in Animation panel)
- AnimationBezier editor (when you have bezier track)
- Any GraphNode-based editor (VisualScript etc.)
If there are no visible regressions (I didn't spot any), this should be ready to merge.

> Have you changed the Sprite converter menu to use this yet?

I looked into it and it has no `gui_input` method, it just uses simple ScrollContainer, so it needs a bit more work. We can gather all editors that need panning improvements and I could do a follow-up to #56756

akien-mga commented:
Thanks!
